app/controlplane/plugins/core/webhook/v1/webhook.go

```diff
@@ -0,0 +1,345 @@
+		i.Logger.Error("missing webhook URL in credentials")
+		return errors.New("missing webhook URL in credentials")
+	}
+	webhookURL := req.RegistrationInfo.Credentials.URL
+
+	// Extract the materials from the attachment state
+	var attachState attachmentState
+	if err := sdk.FromConfig(req.AttachmentInfo.Configuration, &attachState); err != nil {
+		i.Logger.Errorw("invalid attachment state", "error", err)
+		return fmt.Errorf("invalid attachment state: %w", err)
+	}
+
+	// Send attestation if present
+	if req.Input.Attestation != nil {
+		statementJSON, err := json.Marshal(req.Input.Attestation)
+		if err != nil {
+			i.Logger.Errorw("failed to marshal attestation", "error", err)
+			return fmt.Errorf("marshalling attestation: %w", err)
+		}
+		if err := i.sendWebhook(ctx, webhookURL, "ATTESTATION", statementJSON, req.ChainloopMetadata); err != nil {
+			i.Logger.Errorw("failed to send attestation webhook", "error", err)
+			return err
+		}
+	}
+
+	// Send each SBOM if present and specified in the attachment state
+	for _, material := range req.Input.Materials {
+		// Validate material type
+		if material.Type == "" {
+			i.Logger.Warn("encountered material with empty type, skipping")
+			continue
+		}
+
+		// Skip if the material is not in the attachment state
+		if len(attachState.Materials) > 0 && !contains(attachState.Materials, material.Type) {
+			i.Logger.Debugw("material type not attached, skipping", "type", material.Type)
+			continue
+		}
+
+		// Validate material content
+		if len(material.Content) == 0 {
+			i.Logger.Warnw("encountered material with empty content, skipping", "type", material.Type)
+			continue
+		}
+
+		encodedContent := base64.StdEncoding.EncodeToString(material.Content)
+		// create json message with the content
+		jsonMsg := fmt.Sprintf(`{"content": "%s"}`, encodedContent)
+		if err := i.sendWebhook(ctx, webhookURL, material.Type, json.RawMessage(jsonMsg), req.ChainloopMetadata); err != nil {
+			i.Logger.Errorw("failed to send SBOM webhook", "error", err, "type", material.Type)
+			return err
+		}
+	}
+
+	return nil
+}
+
+// sendWebhook sends a webhook with the specified kind and payload
+func (i *Integration) sendWebhook(ctx context.Context, url, kind string, payload json.RawMessage, metadata *sdk.ChainloopMetadata) error {
+	payloadBytes, err := json.Marshal(webhookPayload{
+		Metadata: metadata,
+		Data:     payload,
+		Kind:     kind,
+	})
+	if err != nil {
+		i.Logger.Errorw("failed to marshal webhook payload", "error", err, "kind", kind)
+		return fmt.Errorf("marshalling webhook payload: %w", err)
+	}
+
+	reqHTTP, err := http.NewRequestWithContext(ctx, http.MethodPost, url, bytes.NewBuffer(payloadBytes))
+	if err != nil {
+		i.Logger.Errorw("failed to create HTTP request", "error", err, "url", url)
+		return fmt.Errorf("creating HTTP request: %w", err)
+	}
+	reqHTTP.Header.Set("Content-Type", "application/json")
+
+	resp, err := i.client.Do(reqHTTP)
```

Comment:
you can probably extract the http request logic to a function so it can be used by the validation and for the actual webhook send?